homeassistant/components/motioneye/binary_sensor.py

```diff
@@ -0,0 +1,193 @@
+    """Set up motionEye from a config entry."""
+    entry_data = hass.data[DOMAIN][entry.entry_id]
+
+    @callback
+    def camera_add(camera: dict[str, Any]) -> None:
+        """Add a new motionEye camera."""
+        args = [
+            entry.entry_id,
+            camera,
+            entry_data[CONF_CLIENT],
+            entry_data[CONF_COORDINATOR],
+            entry.options,
+        ]
+        async_add_entities(
+            [
+                MotionEyeMotionBinarySensor(*args),
+                MotionEyeFileStoredBinarySensor(*args),
+            ]
+        )
+
+    listen_for_new_cameras(hass, entry, camera_add)
+
+
+class MotionEyeEventBinarySensor(MotionEyeEntity, BinarySensorEntity):
+    """Base class for motionEye event-based binary sensors."""
+
+    def __init__(
+        self,
+        config_entry_id: str,
+        type_name: str,
+        camera: dict[str, Any],
+        client: MotionEyeClient,
+        coordinator: DataUpdateCoordinator,
+        options: MappingProxyType[str, Any],
+        event: str,
+        friendly_name: str,
+    ) -> None:
+        """Initialize the binary sensor."""
+        super().__init__(
+            config_entry_id, type_name, camera, client, coordinator, options
+        )
+        self._state = False
+        self._event = event
+        self._friendly_name = friendly_name
+        self._timer_unsub: Callable | None = None
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        camera_name = self._camera[KEY_NAME] if self._camera else ""
+        return f"{camera_name} {self._friendly_name}"
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the binary sensor is on."""
+        return self._state
+
+    @callback
+    def _cancel_timer(self) -> None:
+        """Cancel the internal state timer."""
+        if self._timer_unsub is not None:
+            self._timer_unsub()
+            self._timer_unsub = None
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cleanup prior to removal from hass."""
+        self._cancel_timer()
+        await super().async_will_remove_from_hass()
+
+    async def async_added_to_hass(self) -> None:
+        """Register event listeners when added to hass."""
+
+        device_registry = dr.async_get(self.hass)
+
+        @callback
+        def handle_event(event: Event) -> None:
+            """Handle an event."""
+
+            @callback
+            def turn_off(_: datetime.datetime) -> None:
+                """Turn the state off."""
+                self._state = False
+                self.async_write_ha_state()
+                self._timer_unsub = None
+
+            if CONF_DEVICE_ID in event.data:
+                device = device_registry.async_get(event.data[CONF_DEVICE_ID])
+                if device and self._device_identifier in device.identifiers:
+                    self._cancel_timer()
+                    self._timer_unsub = async_call_later(
```

Comment:
We shouldn't offer these entities. They are events and we can't know when to turn them off again. 

If a user wants to turn events into a binary sensor, they should use trigger-based template entities. I have opened a PR to add an example to the docs: https://github.com/home-assistant/home-assistant.io/pull/18567

Reply:
Yes indeed, wasn't sure what the official stance was on these kind of "convenience entities". Not having this will result in a breaking change for those moving from the custom component to core, but I think that's not unreasonable when there's a fairly straightforward documented (thanks!) option.

Adding @frenck for visibility as he had also [suggested such entities](https://github.com/home-assistant/core/pull/49665#pullrequestreview-672382215).

Reply:
Yes, but I didn't intend for timer-based / virtual ones, as that would be indeed simulating a state.